Libki/libki-client
# Improvement for labels defined in configuration file

This request does 2 things.

Firstly, it adds the ability to define custom messages for the different error codes returned by the server. For example, to distinguish between BAD_LOGIN, INVALID_USER and INVALID_PASSWORD. This change permits the use of the different codes as keys in the [labels] section of the configuration file to determine the text to show on the login window if the error arise.

To test:
1. Edit the configuration file to add, under [labels] section:
    INVALID_USER="Custom error message"
or any code/message combination.
2. Launch the client
3. Try to connect with an invalid username
4. Notice the custom message as it appears in the configuration

Secondly, it lets a user set labels' text for multiple languages in the configuration file. It uses a new function that checks through the settings to find the correct one to use depending on the current language. The translations are listed in the configuration file in a section named [labels-<lang>]

To test:
1.  Edit the configuration file to add a new section using the <langcode> of your system's language:
    [labels-<langcode>]
    username="Username <langcode>:"
You can try using the complete code (with the country code), only the language part or an invalid code.
2. Launch the client
3. Notice the right translation is used for the label, using the most precise section when available. Ex:
    i. [labels-fr_CA] if it exists
    ii. [labels-fr] if no i.
    iii. [labels] if neither i. nor ii.
